/**
*  @file  JobResultClient.h
*  @brief  任务分发和任务执行进程间通讯rpc处理
*  @version 0.0.1
*  @since 0.0.1
*  @author
*  @date 2016-7-25  Created it
*/
/******************************************************************************
*@note
Copyright 2007, BeiJing Bluestar Corporation, Limited
ALL RIGHTS RESERVED
Permission is hereby granted to licensees of BeiJing Bluestar, Inc. products
to use or abstract this computer program for the sole purpose of implementing
a product based on BeiJing Bluestar, Inc. products. No other rights to
reproduce, use, or disseminate this computer program,whether in part or  in
whole, are granted. BeiJing Bluestar, Inc. makes no representation or
warranties with respect to the performance of this computer program, and
specifically disclaims any responsibility for any damages, special or
consequential, connected with the use of this program.
For details, see http://www.bstar.com.cn/
******************************************************************************/
#ifndef _TASKRUNNINGRPC_H_
#define _TASKRUNNINGRPC_H_


#include "RpcServer.h"
#include "CustomRpcChannel.h"
#include "RpcCodec.h"
#include "RpcClient.h"

#include "jobdispatch.pb.h"

using namespace vmr::base;
using namespace vmr::protorpc;


namespace vmr
{

	/** @class ImplJobDispatch
	*  @brief  任务分发给任务执行进程的服务端rpc，继承自JobDispatchService
	*   不可拷贝构造和赋值
	*
	*   详见jobdispatch.proto 文件定义
	*/
	class ImplJobDispatch : public JobDispatch::JobDispatchService
	{
	public:
		virtual void add(::google::protobuf::RpcController* controller,
			const ::JobDispatch::TaskRequest* request,
			::JobDispatch::TaskResponse* response,
			::google::protobuf::Closure* done);
		virtual void del(::google::protobuf::RpcController* controller,
			const ::JobDispatch::TaskRequest* request,
			::JobDispatch::TaskResponse* response,
			::google::protobuf::Closure* done);
		virtual void edit(::google::protobuf::RpcController* controller,
			const ::JobDispatch::TaskRequest* request,
			::JobDispatch::TaskResponse* response,
			::google::protobuf::Closure* done);

	public:
		void setPerformContext(void* context)
		{
			performContext_ = context;
		}

	private:
		void* performContext_;
	};

}//namespace 

#endif


